llvm/lib/Target/AMDGPU/AMDGPUCombine.td

```diff
@@ -33,6 +33,11 @@ def rcp_sqrt_to_rsq : GICombineRule<
          [{ return matchRcpSqrtToRsq(*${rcp}, ${matchinfo}); }]),
   (apply [{ Helper.applyBuildFn(*${rcp}, ${matchinfo}); }])>;
 
+def fdiv_by_sqrt_to_rsq : GICombineRule<
+  (defs root:$root),
+  (match (G_FSQRT $sqrt, $x, (MIFlags FmContract)),
+         (G_FDIV $dst, $y, $sqrt, (MIFlags FmContract)):$root),
```

Comment:
Needs more flags. Without the denormal correction code, this isn't more precise. Also switches to dropping denormals and 1ulp. You can refer to the dag version for the exact conditions

Reply:
Also missing hasOneUse check

Reply:
Actually contract is adequate for the f16 case, f32 is the complicated one

Reply:
So I should make a match function to check for conditions for dropping denormals and 1ulp? As for the flags, should I also check for the flags in the same cpp function or add them to the GICombineRule?

Reply:
I am not seeing why / where we need hasOneUse check?

Reply:
Need to check that `G_FSQRT` has one user, so we don't do the combine if it means the `G_FSQRT` stays.
(also means you will need a test with a `G_FSQRT` that has multiple users)

To check `hasOneUse`, you can use a C++ predicate on top of the pattern that calls `MRI.hasOneUse` on `${sqrt}`

Reply:
oddly enough I added the test, and it seemed to work for whatever reason?

Reply:
Do we not want to do the combine at all if the SQRT has multi-uses?

Reply:
Thanks for the reply.